bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/AnnotationUtils.java

```diff
@@ -62,11 +63,40 @@ public static void processAnnotations(MapValue globalAnnotMap, BType bType) {
         }
     }
 
+    /**
+     * Method to retrieve annotations of the type from the global annotation map and set it to the type.
+     *
+     * @param globalAnnotMap The global annotation map
+     * @param bType          The type for which annotations need to be set
+     */
+    public static void processAnnotations_bstring(MapValue globalAnnotMap, BType bType) {
+        if (!(bType instanceof AnnotatableType)) {
+            return;
+        }
+
+        AnnotatableType type = (AnnotatableType) bType;
+        BString annotationKey = StringUtils.fromString(type.getAnnotationKey());
+        if (globalAnnotMap.containsKey(annotationKey)) {
+            type.setAnnotations_bstring((MapValue<BString, Object>) globalAnnotMap.get(annotationKey));
+        }
+
+        if (type.getTag() == TypeTags.OBJECT_TYPE_TAG) {
```

Comment:
inverse if condition here

Reply:
What do you mean by inverse? Am I to use `!=` check?

This is a copy of the previous function to make it work with bstring

Reply:
Can we write like the below ? Will it be more readable with less indentation?

```java
if (type.getTag() != TypeTags.OBJECT_TYPE_TAG) { 
    return;
}

BObjectType objectType = (BObjectType) type;
for (AttachedFunction attachedFunction : objectType.getAttachedFunctions()) {
    annotationKey = StringUtils.fromString(attachedFunction.getAnnotationKey());
    if (!globalAnnotMap.containsKey(annotationKey)) {
        continue;
    }
    attachedFunction.setAnnotations_bstring((MapValue<BString, Object>)
            globalAnnotMap.get(annotationKey));
    }
}
```